src/client.ts

```diff
@@ -105,7 +105,7 @@ export default class Client implements Erdstall {
 			assets,
 		);
 		await tx.sign(this.erdstallConn.erdstall(), this.signer);
-		return this.enclaveConn.transfer(tx);
+		return await this.enclaveConn.transfer(tx);
```

Comment:
Why all those changes to awaits? If a function returns a `Promise` by its signature, better to return a Promise instead of resolving locally. If it's not necessary, please revert all added `awaits`.

Reply:
I had some bugs when transacting or something that were the result of a missing await in an async function. So for good measure, I put awaits everywhere. In my opinion, an async function should always be called with `await`, unless it is intended to be executed in the background.